modules/ww-tritonnp/ww-tritonnp.wdl

```diff
@@ -0,0 +1,190 @@
+  Int map_quality = 20
+  String size_range = "15 500"
+  Int triton_main_ncpus = 4
+
+  # Process each sample
+  scatter (i in range(length(sample_names))) {
+    call triton_main {
+      input:
+        sample_name = sample_names[i],
+        bam_path = bam_paths[i],
+        bam_index_path = bam_index_paths[i],
+        bias_path = bias_paths[i],
+        annotation = annotation,
+        reference_genome = reference_genome,
+        reference_genome_index = reference_genome_index,
+        results_dir = results_dir,
+        map_quality = map_quality,
+        size_range = size_range,
+        cpus = triton_main_ncpus,
+        plot_list = plot_list
+    }
+  }
+
+  # Combine all feature matrices after all samples are processed
+  call combine_fms {
+    input:
+      fm_files = triton_main.fm_file,
+      results_dir = results_dir
+  }
+
+  output {
+    File final_composite = combine_fms.final
+    Array[File] feature_files = triton_main.fm_file
+  }
+}
+
+#### TASK DEFINITIONS ####
+
+# Task definitions file (triton_tasks.wdl)
+task triton_main {
+  meta {
+    description: "Task for running TritonNP"
+    outputs: {
+        fm_files: "Array of outputs from TritonNP"
+    }
+  }
+  parameter_meta {
+    sample_name: "Sample name"
+    bam_path: "Path to BAM file"
+    bam_index_path: "Path to BAM index file"
+    bias_path: "Path to GC corrected file from Griffin"
+    annotation: "Path to BED file of genomic region to process on"
+    reference_genome: "Path to Reference genome file"
+    reference_genome_index: "Path to Reference genome file index"
+    results_dir: "Output directory name"
+    map_quality: "Mapping quality threshold as a positive integer"
+    size_range: "Size range as a tuple, such as 15 500"
+    cpus: "Number of CPUs to use"
+    plot_list: "File containing names of genes to plot."
+  }
+  input {
+    String sample_name
+    File bam_path
+    File bam_index_path
+    File bias_path
+    File annotation
+    File reference_genome
+    File reference_genome_index
+    String results_dir
+    Int map_quality
+    String size_range
+    Int cpus
+    File plot_list
+  }
+
+  command <<<
+    # Create results directory if it doesn't exist
+    mkdir -p ~{results_dir}
+    
+    #Needed Python packges for running Triton
+    python3 -m pip install pysam numpy scipy pandas matplotlib seaborn
+
+    #download script
+    git clone https://github.com/caalo/TritonNP.git
+
+
+    # Run Triton
+    python TritonNP/GenerateFFTFeatures.py \
+      --sample_name ~{sample_name} \
+      --input ~{bam_path} \
+      --bias ~{bias_path} \
+      --annotation ~{annotation}  \
+      --reference_genome ~{reference_genome} \
+      --results_dir ~{results_dir} \
+      --map_quality ~{map_quality} \
+      --size_range ~{size_range} \
+      --cpus ~{cpus} \
+      --plot_list ~{plot_list}
+  >>>
+
+  output {
+    File fm_file = "~{results_dir}/~{sample_name}_PhasingFM.tsv"
+  }
+
+  runtime {
+    cpu: cpus
+    memory: "4 GB" 
```

Comment:
I would add memory as an input parameter if you're allowing CPUs to be adjusted (e.g. 32 CPUs will need more than 4GB of RAM)